apis/kueue/v1beta2/admissioncheck_types.go

```diff
@@ -62,17 +62,23 @@ type AdmissionCheckSpec struct {
 
 type AdmissionCheckParametersReference struct {
 	// apiGroup is the group for the resource being referenced.
+	// +required
 	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
-	APIGroup string `json:"apiGroup"`
+	APIGroup string `json:"apiGroup,omitempty"`
```

Comment:
Why adding omitempty here, since MinLength is 1? (also for Kind and Name fields) .

Reply:
It is to prevent serialization of the empty string when using a structured client.

Reply:
Hm, but why would it matter, empty string is not an allowed value anyway because of the MinLength=1validation. 

Setting required seems to reflect the intention better than optional. However I'm not an API expert so fine if this is the preferred way, it just looks surprising so I'm trying to understand a bit deeper.